qiskit_machine_learning/algorithms/classifiers/vqc.py

```diff
@@ -159,25 +168,118 @@ def num_qubits(self) -> int:
         """Returns the number of qubits used by ansatz and feature map."""
         return self.circuit.num_qubits
 
-    def fit(self, X: np.ndarray, y: np.ndarray):  # pylint: disable=invalid-name
+    def fit(self, X: np.ndarray, y: np.ndarray) -> NeuralNetworkClassifier:
         """
         Fit the model to data matrix X and targets y.
 
         Args:
             X: The input data.
-            y: The target values. Required to be one-hot encoded.
+            y: The target values.
 
         Returns:
-            self: returns a trained classifier.
+            Trained classifier.
         """
-        num_classes = y.shape[-1]
+        y = self._validate_y(y)
+        num_classes = self._get_num_classes(y)
```

Comment:
There have been quite a few changes since you made your comment, but, just to address it now. As far as I can tell the result is preserved in `self._fit_result`, which is only nullified if `warm_start` is `False`. Let me know if I'm missing something though.